core/src/main/scala/akka/persistence/postgres/tag/TagIdResolver.scala

```diff
@@ -13,10 +12,15 @@ trait TagIdResolver {
 
 class CachedTagIdResolver(dao: TagDao, config: TagsConfig)(implicit ctx: ExecutionContext) extends TagIdResolver {
 
+  import akka.persistence.postgres.tag.CachedTagIdResolver.LookupResult
+  import LookupResult._
+
   // TODO add support for loading many tags at once
   // Package private - for testing purposes
-  private[tag] val cache: AsyncLoadingCache[String, Int] =
-    Scaffeine().expireAfterAccess(config.cacheTtl).buildAsyncFuture(findOrInsert(_, config.insertionRetryAttempts))
+  private[tag] val cache: AsyncLoadingCache[String, LookupResult] =
```

Comment:
I am not sure adding this `LookupResult` in the cache itself helps, as per design it cannot contain a `NotFound`.